app/datatables/repository_datatable.rb

```diff
@@ -0,0 +1,346 @@
+          ORDER BY value #{dir} LIMIT #{per_page} OFFSET #{offset}"
+        )
+      else
+        super(records)
+      end
+    else
+      super(records)
+    end
+  end
+
+  # A hack that overrides the new_search_contition method default behavior of the ajax-datatables-rails gem
+  # now the method checks if the column is the created_at and generate a custom SQL to parse
+  # it back to the caller method
+  # def new_search_condition(column, value)
+  #   model, column = column.split('.')
+  #   model = model.constantize
+  #   formated_date = (I18n.t 'time.formats.datatables_date').gsub!(/^\"|\"?$/, '')
+  #   if model == SampleCustomField
+  #     # Find visible (searchable) custom field IDs, and only perform filter
+  #     # on those custom fields
+  #     searchable_cfs = params[:columns].select do |_, v|
+  #       v['searchable'] == 'true' && @cf_mappings.values.include?(v['data'])
+  #     end
+  #     cfmi = @cf_mappings.invert
+  #     cf_ids = searchable_cfs.map { |_, v| cfmi[v['data']] }
+  #
+  #     # Do an ILIKE on 'value', as well as make sure to only include
+  #     # custom fields that have 'custom_field_id' among visible custom fields
+  #     casted_column = ::Arel::Nodes::NamedFunction.new(
+  #       'CAST',
+  #       [model.arel_table[column.to_sym].as(typecast)]
+  #     )
+  #     casted_column = casted_column.matches("%#{value}%")
+  #     casted_column = casted_column.and(
+  #       model.arel_table['custom_field_id'].in(cf_ids)
+  #     )
+  #     casted_column
+  #   elsif column == 'created_at'
+  #     casted_column = ::Arel::Nodes::NamedFunction.new('CAST',
+  #                       [ Arel.sql("to_char( samples.created_at, '#{ formated_date }' ) AS VARCHAR") ] )
+  #     casted_column.matches("%#{sanitize_sql_like(value)}%")
+  #   else
+  #     casted_column = ::Arel::Nodes::NamedFunction.new('CAST',
+  #                       [model.arel_table[column.to_sym].as(typecast)])
+  #     casted_column.matches("%#{sanitize_sql_like(value)}%")
+  #   end
+  # end
+
+  def sort_null_direction(item)
+    val = sort_direction(item)
+    val == 'ASC' ? 'LAST' : 'FIRST'
+  end
+
+  def inverse_sort_direction(item)
+    val = sort_direction(item)
+    val == 'ASC' ? 'DESC' : 'ASC'
+  end
+
+  def sorting_by_custom_column
+    sort_column(params[:order].values[0]) == 'repository_cells.value'
+  end
+
+  # Escapes special characters in search query
+  def escape_special_chars
+    if params[:search].present?
+      params[:search][:value] = ActiveRecord::Base
+                                .__send__(:sanitize_sql_like,
+                                          params[:search][:value])
+    end
+  end
+
+  def new_sort_column(item)
+    coli = item[:column].to_i - 1
+    model, column = sortable_columns[sortable_displayed_columns[coli].to_i]
+                    .split('.')
+
+    return model if model == ASSIGNED_SORT_COL
+    col = [model.constantize.table_name, column].join('.')
```

Comment:
Indeed, is this meant as `coli`? Or should both names be `col`. Is this function used at all?